messages/api.go

```diff
@@ -6,6 +6,37 @@ import (
 	"github.com/Symantec/tricorder/go/tricorder/units"
 )
 
+// Range represents a range for distribution values
+type Range struct {
+	// Represents the lower bound of the range inclusive.
+	// Ignore for the first (lowest) range which never has a lower bound.
+	Lower float64 `json:"lower"`
+	// Represents the upper bound of the range exclusive.
+	// Ignore for the highest (last) range which never has a upper bound.
+	Upper float64 `json:"upper"`
+}
```

Comment:
I see later on that you define buckets using []Range. I would expect that range[n].Upper == range[n+1].Lower, so there appears to be duplication of data. Why not just have []float64 for the bucket endpoints. You would have B+1 endpoints, where B = number of buckets.